pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -553,6 +553,9 @@ func (ds *dockerService) makeSandboxDockerConfig(c *runtimeapi.PodSandboxConfig,
 
 	// Set port mappings.
 	exposedPorts, portBindings := makePortsAndBindings(c.GetPortMappings())
+	if hc.NetworkMode == "none" && len(portBindings) != 0 {
+		return nil, fmt.Errorf("does not support hostport mapping when using cni or kubenet as network plugins")
+	}
```

Comment:
This breaks backward compatibility.
I think we should log a warning rather than return an error.

Reply:
Maybe cni can support hostport in the futher.

Reply:
> This breaks backward compatibility.
I think we should log a warning rather than return an error.

Should log this. But usually users does not care about this log. Keep open on this.